lib/engineering/locations/relative.rb

```diff
@@ -1,14 +1,89 @@
 require 'engineering/locations/base'
+require 'engineering/locations/absolute'
+require 'matrix'
 
 module Engineering
   module Locations
     class Relative < Base
       attr_accessor :value_1, :value_2, :value_3
 
-      def initialize(value_1:, value_2:, value_3:)
+      attr_reader :angle, :origin
+
+      def self.from_location_to_location(from:, to:)
+        array_location = (to.to_matrix - from.to_matrix).to_a
+
+        new(
+          value_1: array_location[0][0],
+          value_2: array_location[1][0],
+          value_3: array_location[2][0],
+          origin: from
+        )
+      end
+
+      def initialize(value_1:, value_2:, value_3:, origin:)
         @value_1 = value_1.to_f
         @value_2 = value_2.to_f
         @value_3 = value_3.to_f
+        @origin = origin
+        @angle = 0.0
+      end
+
+      def align_axis_1_with(vector:)
+        @angle = Math.atan2(vector.value_j, vector.value_i)
+
+        transformed = (transformer_array_global_to_relative * to_matrix).to_a
+        @value_1 = transformed[0][0]
+        @value_2 = transformed[1][0]
+        @value_3 = transformed[2][0]
+
+        self
+      end
+
+      def to_matrix
+        Matrix.column_vector(
+          [
+            value_1,
+            value_2,
+            value_3
+          ]
+        )
+      end
+
+      def to_absolute_location
+        transformed = transformer_array_relative_to_global * to_matrix
+        @angle = 0.0
+
+        initialize_absolute_from_array(matrix: transformed + origin.to_matrix)
+      end
```

Comment:
If this is still needed after the changes we discussed, what do you think about returning an instance of the `Locations::Absolue` class? These `to_*` methods are often used to parse/adapt an instance to something of another class. If this is called `to_absolute_location` I'd expect this to return an instance of `Locations::Absolute`

Reply:
agree